tests/Tests.AzureAppConfiguration/Integration/IntegrationTests.cs

```diff
@@ -0,0 +1,1923 @@
+        {
+            string subscriptionIdFromEnv = Environment.GetEnvironmentVariable("AZURE_SUBSCRIPTION_ID");
+
+            if (!string.IsNullOrEmpty(subscriptionIdFromEnv))
+            {
+                return subscriptionIdFromEnv;
+            }
+
+            // Read the JSON file created by the script
+            string jsonPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "Integration", SubscriptionJsonPath);
+
+            if (!File.Exists(jsonPath))
+            {
+                throw new InvalidOperationException($"Subscription JSON file not found at {jsonPath}. Run the GetAzureSubscription.ps1 script first.");
+            }
+
+            string jsonContent = File.ReadAllText(jsonPath);
+
+            using JsonDocument doc = JsonDocument.Parse(jsonContent);
+            JsonElement root = doc.RootElement;
+
+            bool success = root.GetProperty("Success").GetBoolean();
+
+            if (!success)
+            {
+                throw new InvalidOperationException(root.GetProperty("ErrorMessage").GetString());
+            }
+
+            return root.GetProperty("SubscriptionId").GetString();
+        }
+
+        /// <summary>
+        /// Returns the connection string for connecting to the app configuration store.
+        /// </summary>
+        private string GetConnectionString()
+        {
+            return _connectionString;
+        }
+
+        /// <summary>
+        /// Creates a unique prefix for test keys to ensure test isolation
+        /// </summary>
+        private string GetUniqueKeyPrefix(string testName)
+        {
+            // Use a combination of the test prefix and test method name to ensure uniqueness
+            return $"{TestKeyPrefix}-{testName}-{Guid.NewGuid().ToString("N").Substring(0, 8)}";
+        }
+
+        /// <summary>
+        /// Creates a snapshot with the given name containing the test context's settings
+        /// </summary>
+        private async Task<string> CreateSnapshot(string snapshotName, TestContext testContext)
+        {
+            // Create a snapshot with the test keys
+            var settingsToInclude = new List<ConfigurationSettingsFilter>
+            {
+                new ConfigurationSettingsFilter($"{testContext.KeyPrefix}:*")
+            };
+
+            ConfigurationSnapshot snapshot = new ConfigurationSnapshot(settingsToInclude);
+
+            snapshot.SnapshotComposition = SnapshotComposition.Key;
+
+            try
+            {
+                // Create the snapshot
+                CreateSnapshotOperation operation = await _configClient.CreateSnapshotAsync(WaitUntil.Completed, snapshotName, snapshot);
+
+                // Track created snapshot for cleanup
+                _createdSnapshotNames.Add(snapshotName);
+
+                return operation.Value.Name;
+            }
+            catch (RequestFailedException ex)
+            {
+                Console.WriteLine($"Error creating snapshot: {ex.Message}");
```

Comment:
Avoid catch/log/throw pattern. The exception bubbling up and crashing the tests will create enough logging.